diff --git a/hw/misc/Kconfig b/hw/misc/Kconfig
index ec0fa5aa9f..a91d6c20c9 100644
--- a/hw/misc/Kconfig
+++ b/hw/misc/Kconfig
@@ -25,6 +25,11 @@ config PCI_TESTDEV
     default y if TEST_DEVICES
     depends on PCI
 
+config PCIE_TESTDEVICE
+    bool
+    default y if TEST_DEVICES
+    depends on PCI
+
 config EDU
     bool
     default y if TEST_DEVICES
diff --git a/hw/misc/meson.build b/hw/misc/meson.build
index 6d47de482c..5465c51388 100644
--- a/hw/misc/meson.build
+++ b/hw/misc/meson.build
@@ -157,3 +157,6 @@ system_ss.add(when: 'CONFIG_SBSA_REF', if_true: files('sbsa_ec.c'))
 
 # HPPA devices
 system_ss.add(when: 'CONFIG_LASI', if_true: files('lasi.c'))
+
+# PCIE test device
+system_ss.add(when: 'CONFIG_PCIE_TESTDEVICE', if_true: files('pcie-testdevice.c'))
\ No newline at end of file
